utils/api/benefits/checkAfs.ts

```diff
@@ -0,0 +1,156 @@
+    if (
+      value.livingCountry === LivingCountry.AGREEMENT ||
+      value.everLivedSocialCountry
+    ) {
+      if (meetsReqAge) {
+        return {
+          eligibilityResult: ResultKey.CONDITIONAL,
+          entitlementResult: 0,
+          reason: ResultReason.YEARS_IN_CANADA,
+          detail:
+            "Depending on Canada's agreement with this country, you may be eligible to receive Allowance for Survivor. You are encouraged to contact Service Canada.",
+        }
+      } else if (underAgeReq) {
+        return {
+          eligibilityResult: ResultKey.INELIGIBLE,
+          entitlementResult: 0,
+          reason: ResultReason.AGE,
+          detail:
+            "You may be eligible when you turn 60, depending on Canada's agreement with this country. You are encouraged to contact Service Canada.",
+        }
+      } else {
+        return {
+          eligibilityResult: ResultKey.INELIGIBLE,
+          entitlementResult: 0,
+          reason: ResultReason.AGE,
+          detail:
+            'You must be between 60 and 64 to be eligible for Allowance for Survivor.',
+        }
+      }
+    } else {
+      return {
+        eligibilityResult: ResultKey.INELIGIBLE,
+        entitlementResult: 0,
+        reason: ResultReason.YEARS_IN_CANADA,
+        detail: `You currently do not appear to be eligible for the Allowance for Survivor as you have indicated that you have not lived in Canada for the minimum period of time or lived in a country that Canada has a social security agreement with. However, you may be in the future if you reside in Canada for the minimum required number of years.`,
+      }
+    }
+  } else if (!meetsReqLegal) {
+    if (value.legalStatus === LegalStatus.SPONSORED) {
+      return {
+        eligibilityResult: ResultKey.CONDITIONAL,
+        entitlementResult: 0,
+        reason: ResultReason.LEGAL_STATUS,
+        detail:
+          'You may be eligible for Allowance for Survivor, please contact Service Canada to confirm.',
+      }
+    } else {
+      return {
+        eligibilityResult: ResultKey.CONDITIONAL,
+        entitlementResult: 0,
+        reason: ResultReason.LEGAL_STATUS,
+        detail:
+          'You may be eligible for Allowance for Survivor, and should contact Service Canada to confirm due to your legal status in Canada.',
+      }
+    }
+  } else if (value.livingCountry === LivingCountry.NO_AGREEMENT) {
+    return {
+      eligibilityResult: ResultKey.INELIGIBLE,
+      entitlementResult: 0,
+      reason: ResultReason.SOCIAL_AGREEMENT,
+      detail:
+        'You currently do not appear to be eligible for the Allowance for Survivor as you have indicated that you have not lived in Canada for the minimum period of time or lived in a country that Canada has a social security agreement with. However, you may be in the future if you reside in Canada for the minimum required number of years.',
+    }
+  }
+  // fallback
+  throw new Error('should not be here')
+}
```

Comment:
Looks to be some moved around logic, correct? Not all net new?

Reply:
yeah I moved a lot of stuff around, I think most of the changes in these files is refactoring/cleaning up and not new logic. I tried to organize all the conditions a bit more logically, with the "eligible" path at the top, and the rest is just determining reason for ineligibility.